cmd/init.go

```diff
@@ -77,39 +77,55 @@ var initCmd = &cobra.Command{
-          .ksonnet/      Metadata for ksonnet
-          environments/
-            default/      [Default generated environment]
-          components/    Top-level Kubernetes objects defining application
-          lib/           user-written .libsonnet files
-          vendor/        part libraries, prototypes
+	Long: `Initialize a ksonnet application in a new directory,` + " `app-name`" + `.
+
+This command will generate all the project scafolding required to begin creating and
+deploying components to Kubernetes clusters.
+
+ksonnet applications are initialized based on your current cluster configurations,
+as defined in your` + " `$KUBECONFIG` " + `environment variable. See examples
+of how to customize configurations in the *Examples* section below.
+
+Creating a ksonnet application will result in the following directory tree.
+
+    app-name/
+      .ksonnet/      Metadata for ksonnet
+      app.yaml       Application specifications, ex: name, api version
+      components/    Top-level Kubernetes objects defining the application
+      environments/  Kubernetes cluster definitions
+        default/     Default environment, initialized from the current kubeconfig
+          .metadata/ [1]
+      lib/           user-written .libsonnet files
+      vendor/        part libraries, prototypes
+
+To begin populating your ksonnet application, visit` + " `ks generate` " + `.
+
+[1] Each environment uses a specific version of ksonnet-lib. Users can set flags
+to generate the library based on a variety of data, including server
+configuration and an OpenAPI specification of a Kubernetes build. By default,
+this is generated from the capabilities of the cluster specified in the cluster
+of the current context specified in` + " `$KUBECONFIG`" + `.
 `,
-	Example: `# Initialize ksonnet application, using the capabilities of live cluster
-# specified in the $KUBECONFIG environment variable (specifically: the
-# current context) to generate 'ksonnet-lib'.
+	Example: `# Initialize a ksonnet application, using the live cluster specified in
+# the $KUBECONFIG environment variable (specifically: the current context).
 ks init app-name
 
+# Initialize a ksonnet application, using the context 'dev' from the $KUBECONFIG
+# environment variable. This will initialize the default environment using the
+# server address and default namespace located at the context 'dev'.
+ks init app-name --context=dev
+
+# Initialize a ksonnet application, using the context 'dev' and the namespace
+# 'dc-west' from the $KUBECONFIG environment variable. This will initialize
+# the default environment using the server address at the context 'dev' for
+# the namespace 'dc-west'.
+ks init app-name --context=dev --namespace=dc-west
+
 # Initialize ksonnet application, using the OpenAPI specification generated
 # in the Kubenetes v1.7.1 build to generate 'ksonnet-lib'.
 ks init app-name --api-spec=version:v1.7.1
 
 # Initialize ksonnet application, using an OpenAPI specification file
```

Comment:
Initialize ksonnet application, using the OpenAPI spec generated by a specific build of Kubernetes to...